报错
```json
 {
    "level": "error",
    "ts": 1636541439.3878434,
    "caller": "grpc/reporter.go:74",
    "msg": "Could not send spans over gRPC",
    "error": "rpc error: code = Unavailable desc = last connection error:connection error: desc = \"transport: Error while dialing dial tcp 172.30.0.2:14250: connect: connection refused\"",
    "stacktrace": "github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc.(*Reporter).send\n\t/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc/reporter.go:74\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc.(*Reporter).EmitBatch\n\t/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/reporter/grpc/reporter.go:53\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter.(*MetricsReporter).EmitBatch\n\t/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/reporter/metrics.go:85\ngithub.com/jaegertracing/jaeger/cmd/agent/app/reporter.(*ClientMetricsReporter).EmitBatch\n\t/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/reporter/client_metrics.go:121\ngithub.com/jaegertracing/jaeger/thrift-gen/agent.(*agentProcessorEmitBatch).Process\n\t/go/src/github.com/jaegertracing/jaeger/thrift-gen/agent/agent.go:176\ngithub.com/jaegertracing/jaeger/thrift-gen/agent.(*AgentProcessor).Process\n\t/go/src/github.com/jaegertracing/jaeger/thrift-gen/agent/agent.go:123\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.(*ThriftProcessor).processBuffer\n\t/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:122\ngithub.com/jaegertracing/jaeger/cmd/agent/app/processors.NewThriftProcessor.func2\n\t/go/src/github.com/jaegertracing/jaeger/cmd/agent/app/processors/thrift_processor.go:87"
}
```
- 思路：证书问题
   - 按照 jaeger 官方提供的DockerFile构造 base_image
   - 编译 agent、collector，通过 base_image构建镜像
- 结果
   - 仍然存在验证问题
- 计划
   - 查看grpc机制  